src/user/user.service.ts

```diff
@@ -292,4 +292,30 @@ export class UserService {
       throw new InternalServerErrorException();
     }
   }
+
+    async isUserEmailUnique(id: string, email: string): Promise<boolean> {
```

Comment:
nie wiem czy ta funkcja jest potrzebna bo tak myslę, że sprawdzać unikalność emaila można po prostu używając funkcji findByEmail - jeżeli użytkownik będzie znaleziony to jest to równoznaczne z tym, że email nie jest unikalny :)

Reply:
Wyrzucona i wykorzystana findByEmail :)